src/Symfony/Component/Config/Definition/Builder/ArrayNodeDefinition.php

```diff
@@ -498,7 +517,7 @@ protected function createNode(): NodeInterface
 
         if (isset($this->normalization)) {
             $node->setNormalizationClosures($this->normalization->before);
-            $node->setNormalizedTypes($this->normalization->declaredTypes);
+            $node->setNormalizedTypes($this->allowedTypes ?? $this->normalization->declaredTypes);
```

Comment:
Why is `$this->normalization->declaredTypes` filled normally ? and why do we need to add this new `$this->allowedTypes` property instead of using the existing feature (which gets ignored entirely) ?

Reply:
I was wondering the same until I (re)discovered #44166, which is based on that: sniffing for allowed types in what I'd call an indirect approach. I'm here proposing a more explicit approach, which wins of the implicit one when used.